### ISSUE TITLE
`RL()` method throws a `RuntimeError` when trained on GPU with PyTorch

### ISSUE THREAD

**Issue by AlxndrMlk** [huawei-noah/trustworthyAI#50]
Hi,

thank you for building and sharing such a great package! ❤️

### Issue

I noticed that when using the `RL()` method with GPU (`RL(device_type='gpu')`) I am getting a `RuntimeError` (traceback attached below). 

Other similar methods, like `CORL()` work with GPU without any issues. 

### Setup

I am using `gcastle` version **1.0.3rc2** on **Win 11** with **NVIDIA GeForce GTX 1650** and `torch` backend.

What I understand from the traceback is that some tensors are not being sent to GPU and hence the error.

### Error message:

```{python}
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_32844/3516016318.py in <module>
      3 )
      4 
----> 5 rl.learn(X)

~\anaconda3\envs\econml-dowhy-py38\lib\site-packages\castle\algorithms\gradient\rl\torch\rl.py in learn(self, data, columns, dag, **kwargs)
    258         config.max_length = X.shape[1]
    259 
--> 260         causal_matrix = self._rl(X, config)
    261         self.causal_matrix = causal_matrix
    262 

~\anaconda3\envs\econml-dowhy-py38\lib\site-packages\castle\algorithms\gradient\rl\torch\rl.py in _rl(self, X, config)
    353             if config.device_type == 'gpu':
    354                 reward_feed = callreward.cal_rewards(graphs_feed.cpu().detach().numpy(), lambda1, lambda2)  # np.array
--> 355                 actor.build_reward(reward_ = -torch.from_numpy(reward_feed)[:,0].cuda(config.device_ids))
    356             else:
    357                 reward_feed = callreward.cal_rewards(graphs_feed.detach().numpy(), lambda1, lambda2)  # np.array

~\anaconda3\envs\econml-dowhy-py38\lib\site-packages\castle\algorithms\gradient\rl\torch\models\actor_graph.py in build_reward(self, reward_)
    130         self.reward = reward_
    131 
--> 132         self.build_optim()
    133 
    134     def build_optim(self):

~\anaconda3\envs\econml-dowhy-py38\lib\site-packages\castle\algorithms\gradient\rl\torch\models\actor_graph.py in build_optim(self)
    137         reward_mean, reward_var = torch.mean(self.reward), torch.std(self.reward)
    138         self.reward_batch = reward_mean
--> 139         self.avg_baseline = self.alpha * self.avg_baseline + (1.0 - self.alpha) * reward_mean
    140         if self.config.device_type == 'gpu':
    141             self.avg_baseline = self.avg_baseline.cuda(self.config.device_ids)

RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
```

**Comment by shaido987** [comment]
Hello, 

thank you very much for the error report. We have been able to reproduce the problem and released a new pre-release update with a bug fix (PR #51). Using the new `1.0.3rc3` version of gCastle this should be fixed, you can try and see so that it works for you.

**Comment by AlxndrMlk** [comment]
Hi @shaido987 

yes, it works now. Thank you!
